README.md

```diff
@@ -122,10 +185,108 @@ client.on('connected', async (conn) => {
     conn.failed();
   }
 });
+```
+
+# PulseQueue
+
+A PulseQueue declares a queue and listens for messages on that
+queue, invoking a callback for each messages.
+
+Construct a PulseQueue with the async `consume` function:
 
-client.start();
+```javascript
+let pq = await pulse.consume({
+  client,                // Client object for connecting to the server
+  bindings: [{           // exchange/routingKey patterns to bind to
+    exchange,            // Exchange to bind
+    routingKeyPattern,   // Routing key as string
+    routingKeyReference, // Reference used to parse routing keys (optional)
+  }, ..],
+  handleMessage,         // handler for incoming messages
+  queueName,             // Queue name; if omitted, set exclusiveQueue: true
+  exclusiveQueue,        // If true, use an exclusive queue (and lose messages on reconnect)
+  prefetch,              // Max number of messages unacknowledged to hold (optional)
+  maxLength,             // Maximum queue size, undefined for none
+}
 ```
 
+If `queueName` is specified, this will create a durable queue using a
+pulse-compatible queue name based on `queueName` (prefixed with
+`queue/<namespace>`).  If no `queueName` is given, `exclusiveQueue` must be
+true.
```

Comment:
Testing is going to be done with mocks, anyway.

I don't want to make this library *too* opinionated.  If users want to do something we have not imagined, then getting AssertionError: we don't think you wanted to do that is not really useful.

A named exclusive/autodelete queue makes sense in some cases.  It gets you assurance that there is only one consumer, and that the queue will not fill up when not in use.  This would be good for a command-line tool, for example.

It's worth noting that we're abusing the phrase "exclusive" here to mean "exclusive and autoDelete".  So there's lots of potential sharp edges.